### PR TITLE
Preparing for Puppet-Community release

### PR DESCRIPTION
**Pull request by nibalizer**

### CODE REVIEW

File: metadata.json [hunk 1]
```diff
@@ -1,15 +1,12 @@
 {
-  "name": "puppetlabs-mcollective",
-  "version": "2.0.0",
-  "author": "puppetlabs",
+  "name": "puppet-mcollective",
+  "version": "2.1.0",
+  "author": "puppet",
```
Comment (elyscape) [inline metadata.json]: Should this be `puppetcommunity`?

File: .travis.yml [hunk 2]
```diff
@@ -1,5 +1,8 @@
 ---
+notifications:
+  email: false
 language: ruby
+before_install: rm Gemfile.lock || true
```
Comment (elyscape) [inline .travis.yml]: There's no `Gemfile.lock` in the repository so this line shouldn't be necessary.

Comment (igalic) [inline .travis.yml]: i don't understand why that line isn't simply
```
before_install: rm -f Gemfile.lock
```

Comment (igalic) [inline .travis.yml]: also, can we add
```
sudo: false
```
or will this break anything?

File: Rakefile [hunk 3]
```diff
@@ -12,3 +12,10 @@ task :default do
 end
 
 PuppetLint.configuration.send('disable_80chars')
+
+
+begin
+  require 'puppet_blacksmith/rake_tasks'
+rescue LoadError
+end
+
```
Comment (igalic) [inline Rakefile]: do you wanna copy the the thing i have in extlib, so we can release through travis when we tag?

File: metadata.json [hunk 4]
```diff
@@ -1,15 +1,12 @@
 {
-  "name": "puppetlabs-mcollective",
-  "version": "2.0.0",
-  "author": "puppetlabs",
+  "name": "puppet-mcollective",
+  "version": "2.1.0",
```
Comment (igalic) [inline metadata.json]: why the major version bump?

Comment (nibalizer) [inline metadata.json]: So that we can say 'before 2.0.0 and before was puppetlabs, 2.1.0 and after was puppptcommunity'

File: Rakefile (hunk 3 above)
Comment (nibalizer) [inline Rakefile]: sure

**Comment by igalic** [comment]
@nibalizer any idea why travis is failing?
despite the change to Gemfile to move puppet-blacksmith up, it's not actually happening.

File: .travis.yml [hunk 5]
```diff
@@ -1,5 +1,8 @@
 ---
+notifications:
+  email: false
 language: ruby
+sudo: false
 bundler_args: --without development
```
Comment (igalic) [inline .travis.yml]: add system_tests here too, cuz we're installing fog and all other things we don't need on travis

File: Gemfile [hunk 6]
```diff
@@ -1,16 +1,25 @@
...
+
+group :test do
+  gem "rake"
+  gem "puppet", ENV['PUPPET_VERSION'] || '~> 3.7.0'
+  gem "rspec-puppet", :git => 'https://github.com/rodjek/rspec-puppet.git'
+  gem "puppetlabs_spec_helper"
+  gem "metadata-json-lint"
+  gem "rspec-puppet-facts"
+  gem "rspec", "< 3.2.0", { "platforms" => ["ruby_18"] }
+  gem "puppet-blacksmith", "> 3.3.0", { "platforms" => ["ruby_19", "ruby_20", "ruby_21"] }
```
Comment (igalic) [inline Gemfile]: since our matrix doesn't contain 1.8 we can get rid of the 1.8 hacks.

File: Rakefile [hunk 7]
```diff
@@ -12,3 +12,26 @@ task :default do
 end
 
 PuppetLint.configuration.send('disable_80chars')
+
+
+if RUBY_VERSION >= "1.9.0" then
```
Comment (igalic) [inline Rakefile]: since our matrix doesn't contain 1.8 we can get rid of the 1.8 hacks.

File: Rakefile [hunk 8]
```diff
@@ -6,9 +6,31 @@ Bundler.require :default
...
+desc "Offload release process to Travis."
+task :travis_release => [
+  :check_changelog,  # check that the changelog contains an entry for the current release
+  :"module:release", # do everything except build / push to forge, travis will do that for us
+]
+
+desc "Check Changelog."
+task :check_changelog do
+  v = Blacksmith::Modulefile.new.version
+  if File.readlines('CHANGELOG.md').grep("Releasing #{v}").size == 0 then
```
Comment (igalic) [inline Rakefile]: we'll need to rename our changelog so this matches! (or change this to fit!)

**Pull request merged**